### PR TITLE
feat(beer-encyclopedia): add French legal denomination reference

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Adds the French legal denomination reference table and four nullable
regulatory columns on `Beer`, as the first step of the encyclopedia
data-enrichment plan (PR1 of 3 — Open Food Facts connector and
community-scan loop will land in subsequent PRs). Architectural rationale
and rejected alternatives documented in `ADR-0002`.

- New `legal_denominations` table seeded with the ten denominations
  defined by decree 92-307 of 31 March 1992 modified by 2016-1531
- `Beer.legal_denomination` CHECK-constrained to the same ten codes
  (no FK so the reference table stays re-seedable)
- `Beer.country_of_origin` auto-uppercased ISO 3166-1 alpha-2 (validator
  rejects non-letter input)
- `Beer.allergens` JSONB on PostgreSQL (JSON elsewhere)
- `Beer.alcohol_group` SMALLINT in (1, 3, 4, 5) per Code de la santé
  publique Art. L-3321-1 (group 2 historically merged into 3)
- Idempotent seed script aligned with the existing `seed_styles.py`
  pattern

All four columns are nullable so the migration is non-breaking for
existing rows imported from non-FR sources.

## Checklist

- [x] Happy path + sad path + edge cases covered (32 new tests)
- [x] Migration runs on a fresh SQLite (\`alembic upgrade head\` 001 → 003)
- [x] Seed script idempotent (10 created on first run, 0/10 updated on second)
- [x] \`pytest -q tests/\` green (96/96, no regression)
- [x] \`ruff check\` clean on all new/modified files
- [x] No \`any\`, no inline styles introduced
- [x] ADR-0002 documents the architecture decision

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@vitalikevin — pour information !

Cette PR introduit la première brique de l'enrichissement de la base de données du package `beer-encyclopedia` : un référentiel des dénominations légales françaises de bières (« bière de garde », « pur malt », « panaché »…), tirées du décret 92-307 modifié par le décret 2016-1531.

Concrètement, ça permet maintenant à chaque fiche bière de la base d'indiquer :

- sa **dénomination légale** parmi les 10 catégories officielles du décret
- son **pays d'origine** (code ISO type `FR`, `BE`, `DE`…)
- ses **allergènes** (gluten, sulfites…)
- son **groupe d'alcool** selon le code de la santé publique (l'Art. L-3321-1)

C'est la PR **1 / 3** d'un plan d'enrichissement séquentiel : (1) référentiel légal FR — celle-ci, (2) connecteur Open Food Facts pour ingérer des bières par code-barres EAN, (3) boucle de contribution communautaire via le scan.

Aucun impact côté API NestJS ni côté mobile-app : tout est isolé au package Python. La justification d'architecture (table dédiée vs enum, pas de FK, scope FR-only) est dans l'ADR-0002.

File: packages/beer-encyclopedia/db/models/beer.py [hunk 1]
```diff
@@ -141,3 +182,64 @@ def _validate_source(self, _key: str, value: str) -> str:
...
+        are always two ASCII letters by specification.
+        """
+        if value is None:
+            return value
+        if len(value) != 2 or not value.isalpha():
+            raise ValueError(
+                f"Beer.country_of_origin must be a 2-letter ISO code, "
+                f"got {value!r}"
+            )
+        return value.upper()
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/beer-encyclopedia/db/models/beer.py:245]: Good catch — `str.isalpha()` returns True for Latin-extended letters like `Å` or `Ñ`, which would silently persist invalid country codes. Fixed in follow-up PR #842 (commit `1009d73`): added an `isascii()` guard alongside `isalpha()`, plus a regression test (`test_beer_rejects_non_ascii_letter_country_code`) covering `ÅB` and `ÑO`.
